app/renderer/js/preload.ts

```diff
@@ -79,7 +80,21 @@ process.once('loaded', (): void => {
 // To prevent failing this script on linux we need to load it after the document loaded
 document.addEventListener('DOMContentLoaded', (): void => {
 	if (params.isPageParams()) {
-	// Get the default language of the server
+		const $signInButtons = document.querySelectorAll('.login-social');
```

Comment:
If we go with this approach, it's likely that we'll want to agree on a standard API for this discovery, and add a comment in the webapp codebase forbidding changing the name of the CSS class, so this doesn't do something crazy.  

(I think we also want `.signup-social` for the registration page?)

Reply:
Yeah, we should add that comment too. 
I actually dislike that approach myself, but I figured this would be a quick fix since listening for URL changes causes 2 tabs to open up (since both URLs that Google redirects us through are under the same domain too), and that messes up the local webserver listening for an auth confirmation. 

I'll have to check out the registration page thing, didn't think about that. Thanks!

Reply:
Perhaps this could take adventage of ``external_authentication_methods`` in ``/server_settings`` (https://chat.zulip.org/api/server-settings) like the mobile app to get the urls by which to locate the buttons, instead of hardcoding selection by ``.login-social``, especially since we probably want to change that name, as we've recently merged changes enabling us to render those button for backends that we won't want to call "social". RemoteUser SSO is already included in that.

Reply:
Besides RemoteUser SSO, also SAML is a social backend, but it's not OAuth-based, do these changes account for that?

Reply:
> Perhaps this could take adventage of external_authentication_methods in /server_settings

That's a good idea, I'll have a look at the mobile app's code and change things here. 

> also SAML is a social backend, but it's not OAuth-based, do these changes account for that?

No, I didn't consider SAML when working on this. The problem that we set out to address was mostly with Google logins (since Google has completely blocked all in-app login attempts), so I just jumped to implement an OAuth mechanism to replace our `<webview>` stuff. 
If SAML logins work as is, maybe we can focus on them later?

Reply:
Ah, while it seems like a reasonable thing to do in the future, we should be fine without handling SAML for now. The reason for my question was to make sure the app will know not to apply the OAuth logic to SAML (since it will find SAML in the list ``$signInButtons = document.querySelectorAll('.login-social');``).

Reply:
Yeah, that's a good point, thanks! :)
I think your suggestion of using the `/server_settings` API is great. Once we figure out how to solve this pesky 400 error on the app's end, I'll change the way we intercept requests. That way, OAuth logins won't interfere with SAML ones.